execution/gethexec/executionengine.go

```diff
@@ -53,14 +54,22 @@ import (
 )
 
 var (
-	l1GasPriceEstimateGauge    = metrics.NewRegisteredGauge("arb/l1gasprice/estimate", nil)
-	baseFeeGauge               = metrics.NewRegisteredGauge("arb/block/basefee", nil)
-	blockGasUsedHistogram      = metrics.NewRegisteredHistogram("arb/block/gasused", nil, metrics.NewBoundedHistogramSample())
-	txCountHistogram           = metrics.NewRegisteredHistogram("arb/block/transactions/count", nil, metrics.NewBoundedHistogramSample())
-	txGasUsedHistogram         = metrics.NewRegisteredHistogram("arb/block/transactions/gasused", nil, metrics.NewBoundedHistogramSample())
-	gasUsedSinceStartupCounter = metrics.NewRegisteredCounter("arb/gas_used", nil)
-	blockExecutionTimer        = metrics.NewRegisteredHistogram("arb/block/execution", nil, metrics.NewBoundedHistogramSample())
-	blockWriteToDbTimer        = metrics.NewRegisteredHistogram("arb/block/writetodb", nil, metrics.NewBoundedHistogramSample())
+	l1GasPriceEstimateGauge                = metrics.NewRegisteredGauge("arb/l1gasprice/estimate", nil)
+	baseFeeGauge                           = metrics.NewRegisteredGauge("arb/block/basefee", nil)
+	blockGasUsedHistogram                  = metrics.NewRegisteredHistogram("arb/block/gasused", nil, metrics.NewBoundedHistogramSample())
+	txCountHistogram                       = metrics.NewRegisteredHistogram("arb/block/transactions/count", nil, metrics.NewBoundedHistogramSample())
+	txGasUsedHistogram                     = metrics.NewRegisteredHistogram("arb/block/transactions/gasused", nil, metrics.NewBoundedHistogramSample())
+	txMultiGasUsedComputationHistogram     = metrics.NewRegisteredHistogram("arb/block/transactions/multigasused/computation", nil, metrics.NewBoundedHistogramSample())
+	txMultiGasUsedHistoryGrowthHistogram   = metrics.NewRegisteredHistogram("arb/block/transactions/multigasused/historygrowth", nil, metrics.NewBoundedHistogramSample())
+	txMultiGasUsedStorageAccessHistogram   = metrics.NewRegisteredHistogram("arb/block/transactions/multigasused/storageaccess", nil, metrics.NewBoundedHistogramSample())
+	txMultiGasUsedStorageGrowthHistogram   = metrics.NewRegisteredHistogram("arb/block/transactions/multigasused/storagegrowth", nil, metrics.NewBoundedHistogramSample())
+	txMultiGasUsedL1CalldataHistogram      = metrics.NewRegisteredHistogram("arb/block/transactions/multigasused/l1calldata", nil, metrics.NewBoundedHistogramSample())
+	txMultiGasUsedL2CalldataHistogram      = metrics.NewRegisteredHistogram("arb/block/transactions/multigasused/l2calldata", nil, metrics.NewBoundedHistogramSample())
+	txMultiGasUsedWasmComputationHistogram = metrics.NewRegisteredHistogram("arb/block/transactions/multigasused/wasmcomputation", nil, metrics.NewBoundedHistogramSample())
```

Comment:
1. All these variables could be a single array, just like the multi-gas struct.
2. I believe we want a counter for each dimension instead of a histogram.
3. Given the item above, we can remove `block/transactions` from the path.

The result would be something like:
```
multiGasUsedSinceStartupCounter = make([]metric.Counter, multigas.NumDimensions)

//...

func init() {
  for dimension := multigas.ResourceKind(0); dimension < multigas.NumDimensions; dimension++ {
    // Note we must implement String for ResourceKind, if it is not there already
    metricName := Sprintf("arb/multigas_used/%v", dimension.String().ToLower())
    multiGasUsedSinceStartupCounter[dimension] = metrics.NewRegisteredCounter(metricName, nil)
  }
}
```

Reply:
Thanks, i've just updated the PR with counters